tests/vibe.web.rest.1140/source/app.d

```diff
@@ -0,0 +1,52 @@
+import std.datetime;
+import vibe.d;
+
+interface IOrientDBRoot
+{
+	@property IOrientDBQuery query();
+}
+
+interface IOrientDBQuery
+{
+	@method(HTTPMethod.GET)
+	@path(":db_name/sql/:query/:result_set_size")
+	Json sql(string _db_name, string _query, int _result_set_size);
+}
+
+class OrientDBRoot : IOrientDBRoot {
+	private OrientDBQuery m_query;
+	override @property IOrientDBQuery query() { return m_query; }
+	public this() { this.m_query = new OrientDBQuery(); }
+}
+
+class OrientDBQuery : IOrientDBQuery {
+	override Json sql(string _db_name, string _query, int _result_set_size) {
+		assert(_db_name == Param1, _db_name);
+		assert(_query == Param2, _query);
+		assert(_result_set_size == Param3, to!string(_result_set_size));
+		return Json.emptyObject;
+	}
+}
+
+enum Param1 = "twitter_data";
+enum Param2 = "select DownloadedDateTime from Message";
```

Comment:
I may suggest another one solution (looks like ugly hack=)
You may encode argument at `source/vibe/web/rest.d:1011` and after `Path` will be constructed [here](https://github.com/rejectedsoftware/vibe.d/blob/master/source/vibe/web/rest.d#L373), decode parts separately:
```
auto name_path = Path(name);
immutable(PathEntry)[] name_path_entries;
name_path_entries.reserve(name_path.nodes.length);
foreach(cur; name_path.nodes)
{
    name_path_entries ~= immutable(PathEntry)(filterURLDecode(cur.toString()));
}
url ~= Path(name_path_entries);
``` 
In this case, you also should remove assert from PathEntry [constructor](https://github.com/rejectedsoftware/vibe.d/blob/4070d0645b94c68471109a76b18ba18ececb6399/source/vibe/inet/path.d#L365): now path entry may contain `/` characters. Anyway they will be encoded.